frontend/src/pages/modelServing/screens/global/ModelServingGlobal.stories.tsx

```diff
@@ -143,3 +143,58 @@ DeleteModel.play = async ({ canvasElement }) => {
   await userEvent.type(retypeNameInput, 'Test Inference Service', { delay: 50 });
   expect(deleteButton).not.toBeDisabled();
 };
+
+export const DeployModel = Template.bind({});
+DeployModel.parameters = {
+  a11y: {
+    // need to select modal as root
+    element: '.pf-c-backdrop',
+  },
+};
+DeployModel.play = async ({ canvasElement }) => {
+  // load page and wait until settled
+  const canvas = within(canvasElement);
+  await canvas.findByText('Test Inference Service', undefined, { timeout: 5000 });
+
+  // user flow for editing a project
+  await userEvent.click(canvas.getByText('Deploy model', { selector: 'button' }));
+
+  // get modal
+  const body = within(canvasElement.ownerDocument.body);
+  const nameInput = body.getByRole('textbox', { name: 'Model Name' });
+  const updateButton = body.getByText('Deploy', { selector: 'button' });
+
+  // test that you can not submit on empty
+  await userEvent.clear(nameInput);
+  expect(updateButton).toBeDisabled();
+
+  // fill the form
+  await userEvent.type(nameInput, 'Model Name', { delay: 50 });
+  body.getAllByRole('button', { name: 'Options menu' });
+  await userEvent.click(body.getAllByRole('button', { name: 'Options menu' })[0]);
+  await userEvent.click(body.getByText('Test Project', { selector: 'button' }));
```

Comment:
For selecting items in a dropdown, we should probably use test ids, no? It seems this could be wildly unpredictable if there is any other item on the page saying "Test Project" -- which admittedly, probably never going to happen; but the design leaves some to be desired.